lsp-mode.el

```diff
@@ -7041,8 +7041,8 @@ server. WORKSPACE is the active workspace."
         leftovers body-length body chunk)
     (lambda (_proc input)
       (setf chunk (if (s-blank? leftovers)
-                      input
-                    (concat leftovers input)))
+                      (encode-coding-string input 'utf-8-unix)
```

Comment:
nit: using the 3rd parameters of nocopy might be better due to we don't have to duplicate the string.
Can you check if it still working okay with that?